src/main/java/de/tum/cit/aet/artemis/core/web/admin/AdminOldDataCleanupResource.java

```diff
@@ -0,0 +1,116 @@
+package de.tum.cit.aet.artemis.core.web.admin;
+
+import static de.tum.cit.aet.artemis.core.config.Constants.PROFILE_CORE;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import de.tum.cit.aet.artemis.core.dto.CleanupServiceExecutionRecordDTO;
+import de.tum.cit.aet.artemis.core.security.annotations.EnforceAdmin;
+import de.tum.cit.aet.artemis.core.service.cleanup.DataCleanupService;
+
+/**
+ * REST controller for managing old data cleanup operations in Artemis.
+ * Provides endpoints for administrators to clean up old or orphaned data in the database.
+ */
+@Profile(PROFILE_CORE)
+@RestController
+@RequestMapping("api/admin/cleanup/")
+public class AdminOldDataCleanupResource {
+
+    private static final Logger log = LoggerFactory.getLogger(AdminOldDataCleanupResource.class);
+
+    private final DataCleanupService dataCleanupService;
+
+    public AdminOldDataCleanupResource(DataCleanupService dataCleanupService) {
+        this.dataCleanupService = dataCleanupService;
+    }
+
+    /**
+     * DELETE admin/cleanup/delete-orphans
+     * Deletes orphaned data in the Artemis database.
+     *
+     * @return a {@link ResponseEntity} containing the result of the cleanup operation
+     */
+    @DeleteMapping("delete-orphans")
+    @EnforceAdmin
```

Comment:
Stephan wanted this on the class level (i.e. above the class and not above each method). Did you talk about this because it the same in every other admin resource?